### PR TITLE
Allow GHC 9.8

### LINKED ISSUE #502: Support ghc-9.8.1
This builds with the appropriate `allow-newer` so a Hackage metadata update would be sufficient.
```
> (cd hedgehog && cabal outdated)
Outdated dependencies:
bytestring >=0.10 && <0.12 (latest: 0.12.0.2)
containers >=0.4 && <0.7 (latest: 0.7)
deepseq >=1.1.0.0 && <1.5 (latest: 1.5.0.0)
primitive >=0.6 && <0.9 (latest: 0.9.0.0)
template-haskell >=2.10 && <2.21 (latest: 2.21.0.0)
text >=1.1 && <2.1 (latest: 2.1)
containers >=0.4 && <0.7 (latest: 0.7)
text >=1.1 && <2.1 (latest: 2.1)

> (cd hedgehog-example/ && cabal outdated)
Outdated dependencies:
containers >=0.4 && <0.7 (latest: 0.7)
template-haskell >=2.10 && <2.21 (latest: 2.21.0.0)
text >=1.1 && <2.1 (latest: 2.1)

> (cd hedgehog-dieharder/ && cabal outdated)
Outdated dependencies:
bytestring >=0.10.4.0 && <0.12 (latest: 0.12.0.2)

> (cd hedgehog-test-laws/ && cabal outdated)
Outdated dependencies:
checkers >=0.5 && <0.5.5 (latest: 0.6.0)
tasty >=1.2 && <1.3 (latest: 1.5)
tasty-expected-failure >=0.11 && <0.12 (latest: 0.12.3)
transformers >=0.5 && <0.6 (latest: 0.6.1.1)
```
Let me know if you would prefer a PR.

### PR DESCRIPTION
**Pull request by ysangkok**
Fixes #502

### CODE REVIEW

File: hedgehog/cabal.project [hunk 1]
```diff
@@ -0,0 +1,4 @@
+packages: .
+if impl(ghc >= 9.6)
+  package hedgehog
+    ghc-options: -Wwarn=x-partial
```
Comment (moodmosaic) [inline hedgehog/cabal.project:4]: That’s interesting. Have you kept the error produced without this option?

**Review by moodmosaic — COMMENTED** [review]
LGTM! Just left a comment.

File: hedgehog/cabal.project (hunk 1 above)
Comment (tomjaguarpaw) [reply hedgehog/cabal.project:4]: The error is in the CI build for this PR so it seems the attempt at disabling it failed to take effect for some reason.

```
src/Hedgehog/Internal/Property.hs:429:31: error: [GHC-63394] [-Wx-partial, -Werror=x-partial]
Error:     In the use of ‘head’
    (imported from Hedgehog.Internal.Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists. Use pattern matching or Data.List.uncons instead. Consider refactoring to use Data.List.NonEmpty."
    |
429 |     groups = List.map (\l -> (head l, length l)) $ List.group sp
    |                               ^^^^
```

Comment (tomjaguarpaw) [reply hedgehog/cabal.project:4]: I guess it failed because `cabal.project` was added in a place where it's not being picked up.

Comment (tomjaguarpaw) [reply hedgehog/cabal.project:4]: I submitted an amended PR that fixes this: https://github.com/hedgehogqa/haskell-hedgehog/pull/504

**Pull request closed without merging**